frontend/src/features/rubricBuilder/CSVUpload.tsx

```diff
@@ -1,71 +1,126 @@
-import React from "react";
-import Papa from "papaparse";
-import { CSVRow } from "@local_types";
+import React, { useState } from "react";
+import { importCsv } from "@utils";
+import { Dialog } from "@components";
+
+import { Criteria, Rubric } from "palette-types";
+import { v4 as uuid } from "uuid";
 
 interface CSVUploadProps {
-  onDataChange: (data: CSVRow[]) => void;
-  closeImportCard: () => void; // callback to close the import card
+  rubric: Rubric | undefined;
+  setRubric: React.Dispatch<React.SetStateAction<Rubric | undefined>>;
 }
 
-export const CSVUpload: React.FC<CSVUploadProps> = ({
-  onDataChange,
-  closeImportCard,
-}: CSVUploadProps) => {
-  const handleFileChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    const file = event.target.files?.[0];
-    if (!file) return;
+export const CSVUpload: React.FC<CSVUploadProps> = ({ setRubric }) => {
+  const [showVersionModal, setShowVersionModal] = useState(false);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
 
-    const fileExtension = file.name.split(".").pop()?.toLowerCase();
+  const closeErrorDialog = () => setErrorMessage(null);
 
-    if (fileExtension === "csv") {
-      parseCSV(file);
-    } else {
+  const handleFileChange = (
+    file: File,
+    version: "versionOne" | "versionTwo",
+  ) => {
+    const fileExtension = file.name.split(".").pop()?.toLowerCase();
+    if (fileExtension !== "csv") {
       alert("Unsupported file format. Please upload a CSV file.");
+      return;
     }
+
+    importCsv(
+      file,
+      version,
+      (newCriteria) => {
```

Comment:
could probably just define the callbacks for clarity here

something like

```typescript
const handleImportSuccess = (newCriteria: Criteria[]) => {
  updateRubric(newCriteria);
};

const handleImportError = (error: string) => {
  setErrorMessage(error);
};

importCsv(file, version, handleImportSuccess, handleImportError);

```

Reply:
Created callback for handleImportSuccess and handleImportError